VisualPinball.Unity/VisualPinball.Unity/Game/CameraClipPlane.cs

```diff
@@ -0,0 +1,142 @@
+
+		//Stores the table bounds. 
+		private Bounds tableBounds;
+		private Vector3 previousCameraPosition = Vector3.zero;
+
+		private void OnEnable()
+		{
+			SetClipPlanes(0.001f, 9f); 
+		}
+
+		private void Update()
+		{
+			if(!GetCameraComponent())
+			{
+				return;
+			}
+
+			if(previousCameraPosition != Camera.transform.position)
+			{
+				UpdateClipPlanes();
+				previousCameraPosition = Camera.transform.position; 
+			}
+		}
+
+		/// <summary>
+		/// Updates the camera clipping planes based on the table bounds.
+		/// </summary>
+		public bool UpdateClipPlanes()
+		{
+
+			//Early out if no table is selected.  
+			if(!TableSelector.Instance.HasSelectedTable)
+			{
+				return false;
+			}
+
+			//Early out if we don't have a camera to work on. 
+			if(!GetCameraComponent())
+			{
+				return false;
+			}
+
+			//Get selected table reference. 
+			var table = TableSelector.Instance.SelectedTable; 
+
+			if(Application.isPlaying)
+			{
+				tableBounds = table._tableBounds;  //When playing at runtime, get the stored table bounds value instead of calculating. 
+			}
+			else
+			{
+				tableBounds = table.GetTableBounds();  //When in editor, calculate the bounds in case things have changed.  
+			}
+
+			var trans = Camera.transform;
+
+			var cameraPos = trans.position; // camera position. 
+			var sphereExtent = tableBounds.extents.magnitude; //sphere radius of the bounds. 
+			float cameraToCenter = Vector3.Distance(cameraPos, tableBounds.center);
+			var nearPlane = math.max(cameraPos.magnitude - sphereExtent, 0.001f); //Assign initial near plane used when camera is not in the sphere. 
+			var farPlane = math.max(1f, nearPlane + sphereExtent * 2f); //initial far bounds 
+
+			if(cameraToCenter < sphereExtent)
+			{
+				nearPlane = 0.01f; //camera is in the bounds so drop the near plane to very low. 
+				farPlane = math.max(0.01f, sphereExtent + Vector3.Distance(cameraPos, tableBounds.center)); //set far plane to delta between camera and furthest bound. 
+
+			}
+
+			SetClipPlanes(nearPlane, farPlane); 
+
+			return true; 
+		}
+
+		/// <summary>
+		/// Gets the currently active or attached camera component. 
+		/// </summary>
+		/// <returns>False when no camera can be found.</returns>
+		private bool GetCameraComponent()
+		{
+			Camera = GetComponent<Camera>(); //Get the camera component we are attached to if present.  
+
+			if(Camera == null)
+			{
+				Camera = UnityEngine.Camera.current;  //Get the current active camera if not on the camera component.  
+			}
+
+			if(Camera == null)
+			{
+				return false;
+			}
+
+			return true; 
+		}
+		
+		/// <summary>
+		/// Sets the camera clipping planes based on manually derived values. 
+		/// </summary>
+		/// <param name="near">The near clip plane value</param>
+		/// <param name="far">The far clip plane value</param>
+		/// <returns>False when no camera could be found.</returns>
+		public bool SetClipPlanes(float near, float far)
+		{
+			if(Camera == null) return false; 
```

Comment:
I prefer adding the curly braces because it makes diffing easier, it's more explicit to read, and makes it less error-prone when you need to add more lines to the block (not likely in this example though).

```suggestion
			if (Camera == null) {
				return false;
			}
```